### PR TITLE
Cleanup ServerState.cpp

### PR DESCRIPTION
**Pull request by markuspf**
This PR is a collection that does three (unrelated) cleanups that should not (indeed must not) impact fuctionality.

I could put in separate PRs but that would put load on jenkins and might not be desirable.

 * removes the unused function `mkdir`
 * removes some obsolete comments
 * replaces the use of string literals to access the agency by using string constants

### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [ ] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [ ] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behaviour change can only be verified via automatic tests
 Testing & Verification

This change is a code cleanup without any (additional) test coverage.

### CODE REVIEW

File: arangod/Cluster/ServerState.cpp [hunk 1]
```diff
@@ -47,6 +48,28 @@
...
+using namespace arangodb::consensus;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief helper function to conveniently assemble paths from string constants
+///
+///        this should really be moved to a more appropriate place, and maybe
+///        do checks and/or trims on the strings that are passed in.
+////////////////////////////////////////////////////////////////////////////////
+static std::string concatPath(std::vector<std::string> const& components) {
+  if (components.empty()) {
```
Comment (jsteemann) [inline arangod/Cluster/ServerState.cpp]: the entire method can be replaced by `return StringUtils::join(components, '/');`

File: arangod/Cluster/ServerState.cpp [hunk 2]
```diff
@@ -478,18 +480,14 @@ std::string ServerState::getPersistedId() {
...
 
-static constexpr char const* currentServersRegisteredPref =
-    "/Current/ServersRegistered/";
-
 /// @brief check equality of engines with other registered servers
 bool ServerState::checkEngineEquality(AgencyComm& comm) {
   std::string engineName = EngineSelectorFeature::engineName();
 
-  AgencyCommResult result = comm.getValues(currentServersRegisteredPref);
+  AgencyCommResult result = comm.getValues(concatPath({CURRENT, SERVERS_REGISTERED}));
```
Comment (jsteemann) [inline arangod/Cluster/ServerState.cpp]: The assembled path does not end with a `/` after the change. I guess that isn't a problem, just wanted to be sure this is not an oversight.

File: arangod/Cluster/ServerState.cpp (hunk 1 above)
Comment (markuspf) [reply arangod/Cluster/ServerState.cpp]: Thanks, I did not check `StringUtils.h`, just the `.cpp` file and hence did not see this one.

File: arangod/Cluster/ServerState.cpp (hunk 2 above)
Comment (markuspf) [reply arangod/Cluster/ServerState.cpp]: It is not an oversight; yet I will look & test again that this does not cause a problem.

**Pull request closed without merging**